PhotosExample/ViewController.swift

```diff
@@ -6,14 +6,162 @@
 //
 
 import UIKit
+import Photos
+import SnapKit
 
-class ViewController: UIViewController {
+class ViewController: UIViewController, PHPhotoLibraryChangeObserver {
+    // PHPhotoLibraryChangeObserver: 라이브러리에 변화가 생기면 감지함
 
+    // 가져온 사진들 목록을 tableView에 보여줌
+    private let tableView: UITableView = {
+        let tableview = UITableView()
+        return tableview
+    }()
+    
+    var fetchResult: PHFetchResult<PHAsset>!
+    // imageManager : 가져온 asset을 가지고 이미지를 로드함
+    let imageManager: PHCachingImageManager = PHCachingImageManager()
+    let cellIdentifier: String = "cell"
+  
+    
+    // MARK: 사진 삭제
+    // 테이블 뷰 로우를 밀어서 삭제할 수 있도록 함
+    func tableView(_ tableView: UITableView, canEditRowAt indexPath: IndexPath) -> Bool {
+        return true
+    }
+    
+    // 삭제 모드
+    func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCell.EditingStyle, forRowAt indexPath: IndexPath) {
+        if editingStyle == .delete {
+            let asset: PHAsset  = self.fetchResult[indexPath.row]
+            
+            PHPhotoLibrary.shared().performChanges({
+                PHAssetChangeRequest.deleteAssets([asset] as NSArray)
+            }, completionHandler: nil)
+        }
+        
+    }
+    
+    // 포토 라이브러리가 바뀌면 호출될 메서드
+    func photoLibraryDidChange(_ changeInstance: PHChange) {
+        guard let changes = changeInstance.changeDetails(for: fetchResult)
+        else { return }
+        
+        fetchResult = changes.fetchResultAfterChanges
+        
+        // 바뀌었으면 테이블 뷰 다시 부르기
+        OperationQueue.main.addOperation {
+            self.tableView.reloadSections(IndexSet(0...0), with: .automatic)
+        }
+    }
+    
+    // 카메라 롤 컬렉션 가져오기
+    func requestCollection() {
+        
+        let cameraRoll: PHFetchResult<PHAssetCollection> = PHAssetCollection.fetchAssetCollections(with: .smartAlbum, subtype: .smartAlbumUserLibrary, options: nil)
+        
+        guard let cameraRollCollection = cameraRoll.firstObject else {
+            return
+        }
+        
+        let fetchOptions = PHFetchOptions()
+        // 최신순으로 카메라 sort
+        fetchOptions.sortDescriptors = [NSSortDescriptor(key: "creationDate", ascending: false)]
+        // 그 결과 가져옴
+        self.fetchResult = PHAsset.fetchAssets(in: cameraRollCollection, options: fetchOptions)
+    }
+
+    func setConstraints() {
+        self.view.addSubview(tableView)
+
+        tableView.snp.makeConstraints { make in
+            make.top.bottom.leading.trailing.equalToSuperview()
+        }
+    }
+    
+    // MARK: viewDidLoad
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view.
+        setConstraints()
+        
+        tableView.delegate = self
+        tableView.dataSource = self
+        
+//        tableView.register(cellIdentifier.self, forCellReuseIdentifier: "cell")
```

Comment:
```suggestion
//        tableView.register(PhotoTablesViewCell.self, forCellReuseIdentifier: cellIdentifier)
```
등록하는 것도 tableView 만드는 16번째줄 클로져 안에서 해도 돼요